src/lammps.cpp

```diff
@@ -53,6 +53,12 @@
 #include "memory.h"
 #include "error.h"
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER)
+#if (__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 4))
+#pragma GCC optimize ("O0")
```

Comment:
@akohlmey have you checked to see if this has any performance loss?

Reply:
how could there be any? this only affects the compilation of the body of the LAMMPS class and all it does is create other class instances and handle command line flags.

Reply:
Some compilers support whole-program interprocedural optimization. So turning off all optimizations makes me nervous.

Reply:
1) this applies to GCC only, so it doesn't really matter what other compilers do.
2) this pragma is documented to affect only the statements following it. if GCC would propagate this setting to other files, it would be a bug in GCC.

Reply:
Well:
1. GCC does support both interprocedural optimization and link time optimization, through the flags `-fwhole-program --combine`, and `-flto`. 
2. I'm not saying it would propagate the -O0 setting to other files, but I am saying it might prevent whole-program optimization from occurring.

Is it just a warning you trying to fix by this? This seems like smashing the ant with a pretty big hammer.

Reply:
it is not just the warning, with full optimization it takes an eternity to compile this file for doing something that doesn't need optimization. i will avoid comments on IPO and similar things or else I get yelled at by @sjplimp for being rude in public.

Reply:
Feel free to PM in email then.

Reply:
I think this needs to at least use #pragma GCC push/pop options, or a volatile statement.

Reply:
From the GCC docs:
```
'#pragma GCC push_options'
'#pragma GCC pop_options'

     These pragmas maintain a stack of the current target and
     optimization options.  It is intended for include files where you
     temporarily want to switch to using a different '#pragma GCC
     target' or '#pragma GCC optimize' and then to pop back to the
     previous options.
```

"volatile statement" where?

Reply:
@akohlmey I meant something like this. It compiles very fast, optimizes -O3 everywhere but the single function, and doesn't have the warning:

```patch
diff --git a/src/lammps.cpp b/src/lammps.cpp
index 780327c..adaa2b2 100644
--- a/src/lammps.cpp
+++ b/src/lammps.cpp
@@ -55,8 +55,10 @@

 #if defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER)
 #if (__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 4))
-#pragma GCC optimize ("no-var-tracking-assignments")
+#define OPTFLAG __attribute__((optimize("O0"))) __attribute__((optimize("no-var-tracking-assignments")))
 #endif
+#else
+#define OPTFLAG ""
 #endif

 #include "lmpinstalledpkgs.h"
@@ -902,7 +904,7 @@ void LAMMPS::destroy()
    initialize lists of styles in packages
 ------------------------------------------------------------------------- */

-void LAMMPS::init_pkg_lists()
+void OPTFLAG LAMMPS::init_pkg_lists()
 {
   pkg_lists = new package_styles_lists;
 #define PACKAGE "UNKNOWN"

```

Reply:
if we're going this far, we can also apply it to the equivalent functions in force.cpp and modify.cpp that create instances of styles.

Reply:
@akohlmey sounds good.